done/static/js/src/done.js

```diff
@@ -1,33 +1,51 @@
-function DoneXBlockOn(runtime, element) {
-    $('.done_windshield', element).addClass("done_windshield_off").removeClass("done_windshield_on");
-    DoneXBlock(runtime, element);
+/* Dummy code to make sure events work in Workbench as well as
+ * edx-platform*/
+if (typeof Logger === 'undefined') {
+    var Logger = {
+        log: function(a, b) { return; }
+    }
 }
 
-function DoneXBlockOff(runtime, element) {
-    console.log("b");
-    $('.done_windshield', element).addClass("done_windshield_on").removeClass("done_windshield_off");
-    DoneXBlock(runtime, element);
+function update_knob(element, data) {
+  if($('.done_onoffswitch-checkbox', element).prop("checked")) {
```

Comment:
I agree in most places. In this place, the CSS doesn't have access to the URL of the image (data['checked'] and data['unchecked']). Doing this in JS seemed like a lesser evil than dynamically templating/mucking with the CSS as it came off of the filesystem. I could do the color in CSS, but splitting those two up doesn't make much sense -- the color is just a fallback if the images are slow to load, and kinda related.